File: src/main/java/com/pedigree/model/MediaAttachment.java
package com.pedigree.model;

import java.util.UUID;

public class MediaAttachment {
    private final String id = UUID.randomUUID().toString();
    private String fileName;
    private String relativePath;

    public String getId() { return id; }
    public String getFileName() { return fileName; }
    public void setFileName(String fileName) { this.fileName = fileName; com.pedigree.util.DirtyFlag.setModified(); }
    public String getRelativePath() { return relativePath; }
    public void setRelativePath(String relativePath) { this.relativePath = relativePath; com.pedigree.util.DirtyFlag.setModified(); }
}

File: src/main/kotlin/com/family/tree/model/MediaAttachment.kt
package com.family.tree.model

import com.family.tree.util.DirtyFlag
import java.util.UUID

class MediaAttachment {
    val id: String = UUID.randomUUID().toString()
    var fileName: String? = null
        set(value) { field = value; DirtyFlag.setModified() }
    var relativePath: String? = null
        set(value) { field = value; DirtyFlag.setModified() }
}
